aurora/plot_tools.py

```diff
@@ -151,7 +152,97 @@ def arrow_respond(slider, event):
+        Times of the original data.
+    data : array of float
+        Original data, whose last variable is the time.
+    time_start : float
+        Start time of the simulation.
+    time_end : float
+        End time of the simulation.
+    interval : float
+        Duration of the cycles over which perform the time integration.      
+        
+    Returns
+    -------
+    times : array of float
+        Contains the times of the time-integrated data for all cycles.
+    data_average : array of float
+        Contains the time-integrated data for all cycles.
+    """
+    
+    dt_start = time[0]
+    res = np.linspace(time_start+dt_start,time_end,int((time_end-time_start)/dt_start)) 
+    times = np.linspace(time_start+interval,time_end,int((time_end-time_start)/interval))
+    temp = np.zeros((data.shape[0],data.shape[1],len(res))) 
+    data_average = np.zeros((data.shape[0],data.shape[1],len(times)))
+    
+    for i in range(0,data.shape[0]):
+        for j in range(0,data.shape[1]):
+            f = interp1d(time, data[i,j,:], kind="linear", fill_value="extrapolate", assume_sorted=True)
+            temp[i,j,:] = f(res)
+    
+    for i in range(0,len(times)):
+        idx_lower = find_nearest(res, times[i]-interval)
+        idx_upper = find_nearest(res, times[i])
+        data_average[:,:,i] = np.trapz(temp[:,:,idx_lower:idx_upper],res[idx_lower:idx_upper])/interval
+            
+    return times, data_average
+
+
+def time_average_reservoirs(time,data,time_start,time_end,interval):
+    """Perform time average over cycles of a time-dependent variable.
+
+    Parameters
+    ----------
+    time : array of float
+        Times of the original data.
+    data : array of float
+        Original data, which is a one-dimensional function of the time.
+    time_start : float
+        Start time of the simulation.
+    time_end : float
+        End time of the simulation.
+    interval : float
+        Duration of the cycles over which perform the time integration.      
+        
+    Returns
+    -------
+    times : array of float
+        Contains the times of the time-integrated data for all cycles.
+    data_average : array of float
+        Contains the time-integrated data for all cycles.
+    """
+    
+    dt_start = time[0]
+    res = np.linspace(time_start+dt_start,time_end,int((time_end-time_start)/dt_start)) 
+    times = np.linspace(time_start+interval,time_end,int((time_end-time_start)/interval))
+    temp = np.zeros(len(res))
+    data_average = np.zeros(len(times))
+    
+    f = interp1d(time, data, kind="linear", fill_value="extrapolate", assume_sorted=True)
+    temp = f(res)
+    
+    for i in range(0,len(times)):
+        idx_lower = find_nearest(res, times[i]-interval)
+        idx_upper = find_nearest(res, times[i])
+        data_average[i] = np.trapz(temp[idx_lower:idx_upper],res[idx_lower:idx_upper])/interval
+            
+    return times, data_average
+
+
+def find_nearest(array, value):
```

Comment:
@antonellozito I'd suggest getting rid of this function. One can just do 
```
idx = np.argmin(np.abs(np.asarray(array) - value)))
```
anywhere in the script, wherever it's needed.

Reply:
@fsciortino This is sure a fault of my still poor Python mastering :) I'll do it.